### PR TITLE
:sparkles: Adding structs for the builtin provider file / filecontent

### PR DESCRIPTION
**Pull request by shawn-hurley**

### CODE REVIEW

File: provider/internal/builtin/provider.go [hunk 1]
```diff
@@ -51,14 +51,23 @@ var capabilities = []provider.Capability{
...
+	Filecontent              fileContentCondition `yaml:"filecontent"`
+	File                     fileCondition        `yaml:"file"`
+	XML                      xmlCondition         `yaml:"xml"`
+	JSON                     jsonCondition        `yaml:"json"`
+	HasTags                  []string             `yaml:"hasTags"`
 	provider.ProviderContext `yaml:",inline"`
 }
 
+type fileContentCondition struct {
+	FilePattern string `yaml:"filePattern"`
```
Comment (fabianvf) [inline provider/internal/builtin/provider.go:63]: naming on this one is a little awkward, probably ok though

File: provider/internal/builtin/service_client.go [hunk 2]
```diff
@@ -87,17 +87,28 @@ func (p *builtintServiceClient) Evaluate(cap string, conditionInfo []byte) (prov
...
+				return provider.ProviderEvaluateResponse{}, err
+			}
 		}
 
 		for _, match := range matches {
 			//TODO(fabianvf): This will not work if there is a `:` in the filename, do we care?
 			pieces := strings.SplitN(match, ":", 3)
 			if len(pieces) != 3 {
 				//TODO(fabianvf): Just log or return?
+				//(shawn-hurley): I think the return is good personally
```
Comment (fabianvf) [inline provider/internal/builtin/service_client.go:104]: this is how discussions should happen

**Review by fabianvf — APPROVED** [review]

File: provider/internal/builtin/provider.go (hunk 1 above)
Comment (shawn-hurley) [reply provider/internal/builtin/provider.go:63]: Want to brainstorm?

File: provider/internal/builtin/service_client.go (hunk 2 above)
Comment (shawn-hurley) [reply provider/internal/builtin/service_client.go:104]: I thought it would be good to add, in case someone starts to go through the todo's :)

**Pull request merged**